src/include/blobhash.h

```diff
@@ -14,33 +14,40 @@
 #ifndef CEPH_BLOBHASH_H
 #define CEPH_BLOBHASH_H
 
+#include <cstdint>
 #include "hash.h"
 
-/*
-- this is to make some of the STL types work with 64 bit values, string hash keys, etc.
-- added when i was using an old STL.. maybe try taking these out and see if things 
-  compile now?
-*/
-
 class blobhash {
 public:
-  uint32_t operator()(const char *p, unsigned len) {
-    static rjhash<uint32_t> H;
-    uint32_t acc = 0;
+  uint32_t operator()(const void* p, size_t len) {
```

Comment:
IMHO changing the method's signature is fine from ABI compatibility standpoint as [method derived inside of a class is implicitly `inline`](https://en.cppreference.com/w/cpp/language/inline).